rust/lance-encoding/src/encodings/logical/primitive.rs

```diff
@@ -3346,7 +3346,15 @@ impl PrimitiveFieldEncoder {
                 row_number: 0, // legacy encoders do not use
             })
         })
-        .map(|res_res| res_res.unwrap())
+        .map(|res_res| {
+            res_res.unwrap_or_else(|err| {
+                log::error!("Encoding task failed with error: {:?}", err);
+                Err(lance_core::Error::Internal {
```

Comment:
We don't generally log _and_ raise but it's probably fine here (this should be relatively rare?)